### DISCUSSION TITLE
timed task in experiment

### DISCUSSION THREAD

**Discussion by johnsonjm4** [jspsych/jsPsych#1974] [Q&A]
Hi there,

I am running an experiment that has 3 tasks. One of those tasks is timed. I am trying to make it so that the task doesnt end the rest of the experiment. I have tried using jspsych.endcurrenttimeline() but so far that has ended the rest of the expreriment. here is the important part of code. the rest of it are other tasks. How do I get this to work without ending the entire experiment

var timeline = [];
var specDuration = 10

var matrix_response = {
	type: 'ravens-matrix',
	stimulus: jsPsych.timelineVariable('pattern'),
	data: {task: 'response', test_question: 'test_question', test_condition: jsPsych.timelineVariable('condition'), item: jsPsych.timelineVariable('name'), correct_res: jsPsych.timelineVariable('correct')},
	post_trial_gap: 250,
	choices: [
	jsPsych.timelineVariable('option_01'),
	jsPsych.timelineVariable('option_02'),
	jsPsych.timelineVariable('option_03'),
	jsPsych.timelineVariable('option_04'),
	jsPsych.timelineVariable('option_05'),
	jsPsych.timelineVariable('option_06'),
	jsPsych.timelineVariable('option_07'),
	jsPsych.timelineVariable('option_08')
	],
	on_finish:function(data){
		data.accuracy = jsPsych.pluginAPI.compareKeys(data.correct_res, data.button_pressed)
		psiturk.saveData({
            success: function() {
            	},
        	});  
	}
	on_start: function(){
		setTimeout(function(){
		jsPsych.endCurrentTimeline();
		}, (specDuration*60000));
		}
};

var matrix_proc = {
	timeline: [matrix_response],
	timeline_variables: matrixList
};

**Comment by jodeleeuw** [comment]
Hi @johnsonjm4 ,

How many times is the `matrix_response` task completed? It looks like you are using a setTimeout inside the `on_start` function of the `matrix_response` trial, so if you perform this trial more than one time the `setTimeout` will be set up to occur more than once, which might have the effect of ending other parts of the experiment.
